tasks/apm-inject-install.yml

```diff
@@ -0,0 +1,39 @@
+---
+- name: Include APM host injection Debian install tasks
+  include_tasks: pkg-debian/install-apm-inject-latest.yml
+  when: not ansible_check_mode and ansible_facts.os_family == "Debian" and datadog_apm_instrumentation_enabled != ""
+
+- name: Include APM host injection RedHat install tasks
+  include_tasks: pkg-redhat/install-apm-inject-latest.yml
+  when: not ansible_check_mode and ansible_facts.os_family in ["RedHat", "Rocky", "AlmaLinux"] and datadog_apm_instrumentation_enabled != ""
+
+#- name: Check if dd-host-install needs to run
+#  command: dd-host-install --dry-run
+#  register: datadog_apm_instrumentation_rc
+#  ignore_errors: true
+
+- name: Run APM host injection setup script
+  command: dd-host-install --no-config-change --no-agent-restart
+  notify: restart datadog-agent
+  when: not ansible_check_mode and datadog_apm_instrumentation_rc == 1 and datadog_apm_instrumentation_enabled == "host"
```

Comment:
I'm not sure whether I understand the logic behind the `datadog_apm_instrumentation_rc` variable. Could you please explain it? When would we set it to a different value and how would this all work then?

Reply:
I've changed this to use `register` to the `--dry-run` command execution, please take a look!